mflowgen/tile_array/custom-cts-overrides/cts-overrides.tcl

```diff
@@ -0,0 +1,6 @@
+for {set x 0} {$x < ${array_width}} {incr x} {
```

Comment:
Could you explain what this file is doing?

Reply:
Oops sorry, I forgot to put that in the summary. This page from Cadence support gives an overview of CTS sink types: https://support.cadence.com/apex/ArticleAttachmentPortal?id=a1Od0000000tAwBEAU&pageName=ArticleContent

Basically, we want to prevent CTS from trying to match the clock arrival time for every tile in the array, since we are intentionally skewing the clock down each column. By setting the clock inputs at the top tile of each column as stop sinks, we prevent the tool from looking at clock endpoints down each column during CTS. instead it simply will try to match the arrival time at each stop point (i.e. the top of each column) which is what we want.

Reply:
ah interesting and i guess it's impossible to match the arrival time in every tile, right? because most clock routing is predetermined. so what does it do? just spin for a while?

Reply:
Right, it's not possible in our design. But the problem is actually worse than just unnecessary spin time. Since each of our tiles has 2 clk inputs (one for the tile itself and another to pass on the tile below it). It actually tries to delay the delay the clock arrival to the first row of tiles by inserting a bunch of tiles. I actually saw it add so much delay to the first row that it cause a hold time violation on the data path from the second row of tiles up to the first.